### PR TITLE
Iso refine cleanups 1

### PR DESCRIPTION
**Pull request by drwells**
More followups to #11490.

### CODE REVIEW

File: source/grid/tria.cc [hunk 1]
```diff
@@ -5127,6 +5142,18 @@ namespace internal
              ++quad)
           quad->clear_user_flag();
 
+        // check how much space is needed on every level we need not
```
Comment (masterleinad) [inline source/grid/tria.cc:5145]:
```suggestion
        // check how much space is needed on every level. we need not
```

File: source/grid/tria.cc [hunk 2]
```diff
@@ -5127,6 +5142,18 @@ namespace internal
...
+        // check the highest level since either
+        // - on the highest level no cells are flagged for refinement
+        // - there are, but prepare_refinement added another empty
+        // level which then is the highest level
```
Comment (masterleinad) [inline source/grid/tria.cc:5149]:
```suggestion
        // - there are, but prepare_refinement added another empty
        //   level which then is the highest level
```

File: source/grid/tria.cc [hunk 3]
```diff
@@ -5097,22 +5097,37 @@ namespace internal
...
+                                         std::vector<bool> &vertices_used) {
+          unsigned int next_vertex = current_vertex;
+          while (next_vertex < vertices_used.size() &&
+                 vertices_used[next_vertex] == true)
+            ++next_vertex;
+          Assert(next_vertex < vertices_used.size(), ExcInternalError());
+          vertices_used[next_vertex] = true;
+
+          return next_vertex;
+        };
```
Comment (masterleinad) [inline source/grid/tria.cc:5115]: This is only used much later. Can you move somewhat closer to the place where it is first used?

File: source/grid/tria.cc [hunk 4]
```diff
@@ -5243,7 +5273,27 @@ namespace internal
...
+        for (const auto &cell : triangulation.active_cell_iterators())
+          if (!cell->refine_flag_set())
+            for (unsigned int line_n = 0; line_n < cell->n_lines(); ++line_n)
+              if (cell->line(line_n)->has_children())
+                for (unsigned int c = 0; c < 2; ++c)
+                  Assert(cell->line(line_n)->child(c)->user_flag_set() == false,
+                         ExcInternalError());
+#endif
+
+        unsigned int current_vertex = 0;
```
Comment (masterleinad) [inline source/grid/tria.cc:5297]: Maybe move `get_next_unused_vertex` here?

**Comment by drwells** [comment]
Fixed!

**Review by masterleinad — APPROVED** [review]

**Pull request merged**